CRM/Report/Form/Contribute/Repeat.php

```diff
@@ -400,6 +418,16 @@ public function from() {
        .{$this->contributionJoinTableColumn}
 LEFT JOIN $this->tempTableRepeat2 {$this->_aliases['civicrm_contribution']}2
        ON {$this->groupByTableAlias}.$fromCol = {$this->_aliases['civicrm_contribution']}2.{$this->contributionJoinTableColumn}";
+
+    // for credit card type
+    if ($this->isTableSelected('civicrm_financial_trxn')) {
```

Comment:
This section is repeated so many times I'm inclined to stick it in a function $this->joinFinancialTrxnFromContribution or something.